mlrun/datastore/sources.py

```diff
@@ -412,6 +424,92 @@ def to_dataframe(
             **reader_args,
         )
 
+    def _get_spark_additional_filters(self, column_types: dict):
+        if not self.additional_filters:
+            return None
+        from pyspark.sql.functions import col, isnan, lit
+
+        operators = {
+            "==": operator.eq,
+            "=": operator.eq,
+            ">": operator.gt,
+            "<": operator.lt,
+            ">=": operator.ge,
+            "<=": operator.le,
+            "!=": operator.ne,
+        }
+
+        none_values = [None, np.nan, np.NaN, float("nan")]
+        spark_filter = None
+        new_filter = lit(True)
+        for filter_tuple in self.additional_filters:
+            if not filter_tuple:
+                continue
+            col_name, op, value = filter_tuple
+            if op.lower() in ("in", "not in") and isinstance(value, (list, tuple, set)):
+                none_exists = False
+                value = list(value)
+                for none_value in none_values:
+                    if none_value in value:
+                        value.remove(none_value)
+                        none_exists = True
+                if none_exists:
+                    filter_nan = column_types[col_name] != "timestamp"
+                    if value:
+                        if op.lower() == "in":
+                            new_filter = (
+                                col(col_name).isin(value) | col(col_name).isNull()
+                            )
+                            new_filter = (
+                                new_filter | isnan(col(col_name))
+                                if filter_nan
+                                else new_filter
+                            )
+                        else:
+                            new_filter = (
+                                ~col(col_name).isin(value) & ~col(col_name).isNull()
+                            )
+                            new_filter = (
+                                new_filter & ~isnan(col(col_name))
+                                if filter_nan
+                                else new_filter
+                            )
```

Comment:
Simplify
```suggestion
                            if filter_nan:
                                new_filter = new_filter & ~isnan(col(col_name))
```
(all occurrences)